### PR TITLE
feat(*): add OSM flag logic

### PR DESCRIPTION
**Pull request by shalier**
Renames `clustertype` to more generic `flag` and adds the OSM flag and logic

### CODE REVIEW

File: cmd/aks-periscope/aks-periscope.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func main() {
 		log.Fatalf("Failed to create CRD: %v", err)
 	}
 
-	clusterType := os.Getenv("CLUSTER_TYPE")
+	flags := strings.Fields(os.Getenv("FLAG_LIST"))
```
Comment (SanyaKochhar) [inline cmd/aks-periscope/aks-periscope.go]: could this be named something even more specific like `collectorFlags` ?

File: deployment/aks-periscope.yaml [hunk 2]
```diff
@@ -143,10 +143,10 @@ data:
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: clustertype-config
+  name: flag-config
   namespace: aks-periscope
 data:
-  CLUSTER_TYPE: "managedCluster" # <custom flag>
+  FLAG_LIST: # <custom flag> connectedCluster OSM
```
Comment (SanyaKochhar) [inline deployment/aks-periscope.yaml]:
```suggestion
  FLAG_LIST: # <custom flag> "connectedCluster" "OSM"
```
does this need to be a string in quotes based on the prev example? shouldn't matter right? I think yaml should be able to render it?

**Review by SanyaKochhar — APPROVED** [review]
just a tiny nit about naming

File: deployment/aks-periscope.yaml (hunk 2 above)
Comment (shalier) [reply deployment/aks-periscope.yaml]: @SanyaKochhar Yeah it doesn't matter if the string is wrapped in quotes or not the yaml can detect that it's a string

Comment (SanyaKochhar) [inline deployment/aks-periscope.yaml]: could this be renamed to `collectorFlags` as well?

**Pull request merged**